public/js/review-searchBar.js

```diff
@@ -0,0 +1,125 @@
+import { CLASS_LIST } from './util/cssClasses.js';
+import { delay } from './util/util.js';
+import { URL, KEYCODE } from './util/data.js';
+
+class SearchBar {
+  constructor({ domElem, inputArea, suggestions, rollingKeywords }) {
+    this.domElem = domElem;
+    this.inputArea = inputArea;
+    this.suggestions = suggestions;
+    this.rollingKeywords = rollingKeywords;
+  }
+
+  registerEvent() {
+    const { HIDDEN } = CLASS_LIST;
+
+    this.inputArea.addEventListener('focus', () => {
+      this.suggestions.classList.remove(HIDDEN);
+      this.rollingKeywords.classList.add(HIDDEN);
+    })
+
+    this.domElem.addEventListener('mouseleave', () => {
+      if(this.inputArea.value === '') {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
+        this.rollingKeywords.classList.remove(HIDDEN);
+      }
+      if(this.inputArea.value) {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
+      }
+    });
+  }
+}
+
+class Rolling {
+  constructor({ domElem, rollingData }) {
+    this.domElem = domElem;
+    this.rollingData = rollingData;
+    this.currItemIdx = 0;
+    this.currY = 4;
+    this.heightOfOneItem = 30;
+    this.init();
+  }
+
+  render() {
+    let renderingValue = this.rollingData.reduce((prev, curr, idx) => {
+      return prev + `<li><span class="num-rank">${idx + 1}</span>${this.rollingData[idx]}</li>`
+    }, '')
+    renderingValue += `<li><span class="num-rank">1</span>${this.rollingData[0]}</li>`
+    this.domElem.innerHTML = renderingValue;
+  }
+
+  async roll() {
+    await delay('', 2000);
+    const rollngTimeoutId = setTimeout(() => {
+      this.setAnimation();
+      this.currItemIdx++;
+      if(this.currItemIdx === 11) {
+        this.resetStates();
+      }
+      this.roll()
+    }, 1000);
```

Comment:
여기서 사용되는 setTimeout도 delay함수로 대신할 수 있지 않을까요? 그렇게 되면 위에서 호출하는 delay함수가 3000ms을 지연시키는 것이 좋겠네요 🤔 
```suggestion
    await delay(1000);
    this.setAnimation();
    this.currItemIdx++;
    if(this.currItemIdx === 11) {
      this.resetStates();
    }
    this.roll();
```

Reply:
delay함수를 만들어놓고 왜 쓸 생각을 못했을까요? 알려주셔서 감사합니다 ㅠㅠㅠㅠㅠㅠㅠ!!!!